src/MissingToken.php

```diff
@@ -6,11 +6,14 @@
 
 namespace Microsoft\PhpParser;
 
+use ReturnTypeWillChange;
+
 class MissingToken extends Token {
     public function __construct(int $kind, int $fullStart) {
         parent::__construct($kind, $fullStart, $fullStart, 0);
     }
 
+    #[ReturnTypeWillChange]
     public function jsonSerialize() {
```

Comment:
https://wiki.php.net/rfc/internal_method_return_types adds support for tentative return types, and `:mixed` was later added to JsonSerializable's method in 8.1 in https://github.com/php/php-src/pull/7051 . This means that implementations will have to return something (or throw), and also that the absence of a return type will emit a warning.

- fixes `vendor/microsoft/tolerant-php-parser/src/Token.php:113 [8192] Declaration of Microsoft\PhpParser\Token::jsonSerialize() should be compatible with JsonSerializable::jsonSerialize(): mixed` (a warning, not an error)

Because php 7 is still supported, add an annotation to suppress the warning instead.

Eventually this may have to drop support for 7.x because the mixed type requires 8.0, but that's probably enough time, but that depends on how many 8.x minor releases there are, which is unknown